src/Nethermind/Nethermind.Synchronization/FastBlocks/ReceiptsSyncFeed.cs

```diff
@@ -75,6 +91,26 @@ public override void InitializeFeed()
                 _barrier = _syncConfig.AncientReceiptsBarrierCalc;
                 if (_logger.IsInfo) _logger.Info($"Changed pivot in receipts sync. Now using pivot {_pivotNumber} and barrier {_barrier}");
                 ResetSyncStatusList();
+                if (!_receiptStorage.HasBlock(_syncConfig.PivotNumberParsed, _syncConfig.PivotHashParsed))
+                {
+                    _barrierWhenStarted = _syncConfig.AncientReceiptsBarrierCalc;
+                    _metadataDb.Set(MetadataDbKeys.ReceiptsBarrierWhenStarted, _barrierWhenStarted.Value.ToBigEndianByteArrayWithoutLeadingZeros());
+                }
+                else if (_metadataDb.KeyExists(MetadataDbKeys.ReceiptsBarrierWhenStarted))
+                {
+                    _barrierWhenStarted = _metadataDb.Get(MetadataDbKeys.ReceiptsBarrierWhenStarted).ToLongFromBigEndianByteArrayWithoutLeadingZeros();
+                }
+                else if (_specProvider.ChainId == BlockchainIds.Mainnet)
+                {
+                    // Assume the receipts barrier was the previous defualt (deposit contract barrier) only for mainnet
+                    _barrierWhenStarted = DepositContractBarrier;
```

Comment:
Edge case (not sure if worth handling): if the mainnet user with config barriers different than default deposits will upgrade Nethermind during receipts download it will stop receipts sync.

Reply:
I would ignore case when someone upgrades during sync